### DISCUSSION TITLE
How to build a dynamic secondary side menu based on back-end interface data

### DISCUSSION THREAD

**Discussion by whg517** [refinedev/refine#4898] [Q&A] [answered]
I now have a request, in the left menu bar there is a service menu, its subset is a number of specific services such as (hive, trino, vault, etc.), and can be clicked to view.

When the front-end interface is opened after the initial installation of the system, there is no second-level menu for services. Through the installation process, the back-end will install the selected services into the cluster environment. At this time, the second-level menu of the service has the newly installed services. The service list is dynamically displayed by obtaining persistent data from the back end through the interface.

eg:

![image](https://github.com/refinedev/refine/assets/21030747/cea91995-46cd-46eb-9f9e-ef98274c844c)

**Comment by alicanerdurmaz** [comment, marked as answer]
Hello @whg517,

We have 2 options.
1. You can create a custom `<Sider />` from scratch according to your needs.
2. You can create dynamic `resources` with [Multi Level menu](https://refine.dev/docs/advanced-tutorials/multi-level-menu/#what-is-multi-level-menu).

**Reply by whg517** [reply]
got it, i will try , tks
